File: dotnet/Eth/Libs/Common/Dapr.Abstracts/IStateManager.cs
using Dapr.Client;

namespace Dapr.Abstracts
{
    public interface IStateEntry<TValue>
    {
        string Key { get; }
        TValue Value { get; set; }
        string ETag { get; }
    }

    public interface IStateManager
    {
        Task<bool> TrySave<TValue>(string storeName, string key, TValue value, string etag, StateOptions? stateOptions = null, IReadOnlyDictionary<string, string>? metadata = null, CancellationToken cancellationToken = default);
        Task Save<TValue>(string storeName, string key, TValue value, StateOptions? stateOptions = null, IReadOnlyDictionary<string, string>? metadata = null, CancellationToken cancellationToken = default);
        Task<IStateEntry<TValue>> GetEntry<TValue>(string storeName, string key, ConsistencyMode? consistencyMode = default, IReadOnlyDictionary<string, string>? metadata = default, CancellationToken cancellationToken = default);
        Task<TValue> Get<TValue>(string storeName, string key, ConsistencyMode? consistencyMode = default, IReadOnlyDictionary<string, string>? metadata = default, CancellationToken cancellationToken = default);
        Task Delete(string storeName, string key, StateOptions? stateOptions = default, IReadOnlyDictionary<string, string>? metadata = default, CancellationToken cancellationToken = default);
    }

}

File: dotnet/Eth/Libs/Common/DaprState/IStateManager.fs
namespace Common.DaprState

open Dapr.Client

  type IValue<'TValue> =
    abstract ETag: string
    abstract Value: 'TValue

  type IStateManager =
    abstract trySave:
      storeName: string *
      key: string *
      value: 'TValue *
      etag: string *
      ?stateOptions: StateOptions *
      ?metadata: System.Collections.Generic.IReadOnlyDictionary<string, string> *
      ?cancellationToken: System.Threading.CancellationToken ->
        System.Threading.Tasks.Task<bool>

    abstract save:
      storeName: string *
      key: string *
      value: 'TValue *
      ?stateOptions: StateOptions *
      ?metadata: System.Collections.Generic.IReadOnlyDictionary<string, string> *
      ?cancellationToken: System.Threading.CancellationToken ->
        System.Threading.Tasks.Task

    abstract getEntry:
      storeName: string *
      key: string *
      ?consistencyMode: System.Nullable<ConsistencyMode> *
      ?metadata: System.Collections.Generic.IReadOnlyDictionary<string, string> *
      ?cancellationToken: System.Threading.CancellationToken ->
        System.Threading.Tasks.Task<IValue<'TValue>>

    abstract get<'TValue> :
      storeName: string *
      key: string *
      ?consistencyMode: System.Nullable<ConsistencyMode> *
      ?metadata: System.Collections.Generic.IReadOnlyDictionary<string, string> *
      ?cancellationToken: System.Threading.CancellationToken ->
        System.Threading.Tasks.Task<'TValue>

    abstract delete:
      storeName: string *
      key: string *
      ?stateOptions: StateOptions *
      ?metadata: System.Collections.Generic.IReadOnlyDictionary<string, string> *
      ?cancellationToken: System.Threading.CancellationToken ->
        System.Threading.Tasks.Task
